packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -295,22 +295,42 @@ class RenderEditable extends RenderBox {
   ValueListenable<bool> get selectionEndInViewport => _selectionEndInViewport;
   final ValueNotifier<bool> _selectionEndInViewport = ValueNotifier<bool>(true);
 
+  // Returns a bool indicating if the offset is within factor of being contained
+  // within the rect.
+  static bool _containsFuzzy(Rect rect, Offset offset, [double factor = 0.1]) {
+    return rect.contains(offset)
+      || rect.contains(offset + Offset(factor, factor))
+      || rect.contains(offset + Offset(factor, -factor))
+      || rect.contains(offset + Offset(-factor, factor))
+      || rect.contains(offset + Offset(-factor, -factor));
+  }
+
   void _updateSelectionExtentsVisibility(Offset effectiveOffset) {
     final Rect visibleRegion = Offset.zero & size;
 
     final Offset startOffset = _textPainter.getOffsetForCaret(
       TextPosition(offset: _selection.start, affinity: _selection.affinity),
       Rect.zero,
     );
-
-    _selectionStartInViewport.value = visibleRegion.contains(startOffset + effectiveOffset);
+    // Check if the caret is visible with an approximation because of an error
+    // in the engine that always reports that the caret extends above the input
+    // by a small amount. See paragraph.cc's implementation of GetRectsForRange.
+    const double kApproximateFactor = 0.1;
```

Comment:
For const variables, don't use the "k" prefix. In this case (sadly) the variable name that would be consistent with existing names would be visibleRegionSlop.